sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/ExceptionExtensions.cs

```diff
@@ -70,6 +70,9 @@ internal static class ExceptionExtensions
                 case TaskCanceledException _:
                     return new EventHubsException(true, eventHubName, Resources.CouldNotCreateLink, EventHubsException.FailureReason.ServiceCommunicationProblem, instance);
 
+                case ObjectDisposedException _:
+                    return new EventHubsException(false, eventHubName, Resources.CouldNotCreateLink, EventHubsException.FailureReason.ClientClosed, instance);
```

Comment:
Interesting, I have a work item to actually remove the ClientClosed reason and just always throw ObjectDisposedException directly whether originating from client closing or something else.
https://github.com/Azure/azure-sdk-for-net/issues/12856

I guess we can't do this same change in EH, but just wanted to put it out there that this was the tentative plan for SB.

Reply:
I think you're good in SB.  Either way seems reasonable to me.  At this point, I don't want to change the GA behavior.   I added this here to normalize because the logic had been isolated to just the background publishing.

For context, we ended up this way for two reasons:

- Since the client doesn't technically need to be disposed, it seemed misleading to allow an `ObjectDisposed` to leak out if the client had been closed but may not have been disposed. 

- There was some concern about confusion, because the [Exception documentation](https://docs.microsoft.com/en-us/azure/event-hubs/event-hubs-messaging-exceptions) calls out that `OperationCanceledException` gets thrown when you do something to a disposed object.